Add unit tests for `containerd` and `cri` Salt module

TeddyAndrieux opened the pull request:
**Component**:

'salt,' 'tests'

**Context**: 

#2266 

**Summary**:

- Add tests utils to simulate  salt `cmd.run_all` output
- Small fix in `cri.execute`
- Add unit tests for `containerd` and `cri` custom Salt module

---

Refs: #2266

TeddyAndrieux committed:
salt/tests: Add unit tests for `containerd` salt module

Add unit tests for `containerd` salt custom module

Refs: #2266

TeddyAndrieux committed:
salt/tests: Add simple utils to get cmd `run_all` output

Add a simple helper to get a simple `cmd.run_all` salt output

TeddyAndrieux committed:
salt: Check `container_id` before execute in `cri.execute`

`crictl ps` return empty result with retocde 0 when the container does
not exists, so add a check in `cri.execute` custom Salt module before
executing the command, to check that container_id is not empty

TeddyAndrieux committed:
salt/tests: Add unit tests for `cri` salt module

Add unit tests for `cri` salt custom module

Refs: #2266

bert-e commented:
# Hello teddyandrieux,



My role is to assist you with the merge of this
pull request. Please type `@bert-e help` to get information
on this process, or consult the [user documentation](
https://eve.devsca.com/github/scality/metalk8s/bert-e/doc/user).





*Status report is not available.*

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/GH-2266-add-salt-module-unit-test` into
`development/2.4`
* [w/2.5/improvement/GH-2266-add-salt-module-unit-test](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/GH-2266-add-salt-module-unit-test) will be merged into `development/2.5` 
* [w/2.6/improvement/GH-2266-add-salt-module-unit-test](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/improvement/GH-2266-add-salt-module-unit-test) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality

TeddyAndrieux committed:
salt/tests: Add unit test for `__virtual__` of `metalk8s_network`

Add a unit test for `__virtual__` function of `metalk8s_network` custom
Salt modules so that we have 100% coverage on this Salt module

Sees: #2266

bert-e commented:
# History mismatch



Merge commit #e92cc046774ac26e58d9762a6aff7631e44362bd on the integration branch
`w/2.5/improvement/GH-2266-add-salt-module-unit-test` is merging a branch which is neither the current
branch `improvement/GH-2266-add-salt-module-unit-test` nor the development branch
`development/2.5`.

It is likely due to a rebase of the branch `improvement/GH-2266-add-salt-module-unit-test` and the
merge is not possible until all related `w/*` branches are deleted or updated.

**Please use the `reset` command to have me reinitialize these branches.**

TeddyAndrieux commented:
/reset

bert-e commented:
# Reset complete



I have successfully deleted this pull request's integration branches.

bert-e commented:
# Integration data created



I have created the integration data for the additional destination branches.

* this pull request will merge `improvement/GH-2266-add-salt-module-unit-test` into
`development/2.4`
* [w/2.5/improvement/GH-2266-add-salt-module-unit-test](https://github.com/scality/metalk8s/compare/development/2.5...w/2.5/improvement/GH-2266-add-salt-module-unit-test) will be merged into `development/2.5` 
* [w/2.6/improvement/GH-2266-add-salt-module-unit-test](https://github.com/scality/metalk8s/compare/development/2.6...w/2.6/improvement/GH-2266-add-salt-module-unit-test) will be merged into `development/2.6` 



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`




You can set option `create_pull_requests` if you need me to create
**integration pull requests** in addition to integration branches, with:

```
@bert-e create_pull_requests
```

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality

Ebaneck commented on salt/tests/unit/modules/test_cri.py:
```diff
@@ -0,0 +1,194 @@
+import cri
+
+
+IMAGES_LIST = [{
```
Since we are starting to have variables of 21+ lines, what is the good practice here...
 - To define a global helper file that contains these variables and simply import them from there or
 - Move the variable declaration to the bottom of this file

TeddyAndrieux replied:
To me I think it's good to have example of output you get from command in test so that it's easier to understand the code, anyway to me here it's still simple ones so it's ok to have them directly in the python test file IMHO.

For complicated ones I move to file check https://github.com/scality/metalk8s/pull/2656

Ebaneck reviewed (commented):
Seems ok for me...
cc @alexandre-allard-scality

TeddyAndrieux commented:
/approve

bert-e commented:
# Waiting for approval



The following approvals are needed before I can proceed with the merge:


* the author


* one peer





Peer approvals *must* include at least 1 approval from the following list:

* @jbertran

* @tcarmet

* @thomasdanan

* @sayf-eddine-scality

* @lucieleonard

* @monpote

* @carlito-scality

* @cmonfort

* @nicolast

* @slaperche-scality

* @teddyandrieux

* @nootal

* @gdemonet

* @alexandre-allard-scality










*The following options are set:* **approve**

bert-e commented:
# In the queue



The changeset has received all authorizations and has been added to the
relevant queue(s). The queue(s) will be merged in the target development
branch(es) as soon as builds have passed.

The changeset will be merged in:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches will **NOT** be impacted:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



There is no action required on your side. You will be notified here once
the changeset has been merged. In the unlikely event that the changeset
fails permanently on the queue, a member of the admin team will
contact you to help resolve the matter.

*IMPORTANT*

Please do not attempt to modify this pull request.

* Any commit you add on the source branch will trigger a new cycle after the
  current queue is merged.
* Any commit you add on one of the integration branches will be **lost**.

If you need this pull request to be removed from the queue, please contact a
member of the admin team now.




*The following options are set:* **approve**

bert-e commented:
# 


I have successfully merged the changeset of this pull request
into targetted development branches:


* :heavy_check_mark: `development/2.4`

* :heavy_check_mark: `development/2.5`

* :heavy_check_mark: `development/2.6`



The following branches have **NOT** changed:

* `development/1.0`
* `development/1.1`
* `development/1.2`
* `development/1.3`
* `development/2.0`
* `development/2.1`
* `development/2.2`
* `development/2.3`



Please check the status of the associated issue GH-2266.

Goodbye teddyandrieux.
